dataset_eval/evaluate.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import print_function
+
+import sys
+import os
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), ".."))
+import config
+
+from dataset_eval import gaia_wrapper
+import gaia2.fastyaml as yaml
+import db
+import db.data
+import db.dataset
+import db.dataset_eval
+import db.exceptions
+from db.utils import create_path
+import unicodedata
+import tempfile
+import shutil
+import time
+import json
+import os
+import re
+
+SLEEP_DURATION = 30  # number of seconds to wait between runs
+
+
+def main():
+    print("Starting dataset evaluator...")
+    while True:
+        db.init_db_connection(config.PG_CONNECT)
+        pending_job = db.dataset_eval.get_next_pending_job()
+        if pending_job:
+            print("Processing job %s..." % pending_job["id"])
+            evaluate_dataset(pending_job)
+        else:
+            print("No pending datasets. Sleeping %s seconds." % SLEEP_DURATION)
+            db.close_connection()
+            time.sleep(SLEEP_DURATION)
+
+
+def evaluate_dataset(eval_job):
+    db.dataset_eval.set_job_status(eval_job["id"], db.dataset_eval.STATUS_RUNNING)
+    temp_dir = tempfile.mkdtemp()
+
+    try:
+        dataset = db.dataset.get(eval_job["dataset_id"])
+
+        filelist_path = os.path.join(temp_dir, "filelist.yaml")
+        filelist = dump_lowlevel_data(extract_recordings(dataset), os.path.join(temp_dir, "data"))
+        with open(filelist_path, "w") as f:
+            yaml.dump(filelist, f)
+
+        groundtruth_path = os.path.join(temp_dir, "groundtruth.yaml")
+        with open(groundtruth_path, "w") as f:
+            yaml.dump(create_groundtruth(dataset), f)
+
+        results = gaia_wrapper.train_model(
+            groundtruth_file=groundtruth_path,
+            filelist_file=filelist_path,
+            project_dir=temp_dir,
+        )
+        db.dataset_eval.set_job_result(eval_job["id"], json.dumps(results))
+        db.dataset_eval.set_job_status(eval_job["id"], db.dataset_eval.STATUS_DONE)
+
+    # TODO(roman): Also need to catch exceptions from Gaia.
```

Comment:
Gaia exceptions are pretty important. Can we hack this for now? Some ideas:

 - get the stacktrace and look for "gaia2"
 - find a list of common exceptions and check `msg`